service/controller.go

```diff
@@ -3130,7 +3140,8 @@ func (s *service) SelectOrCreateFCPGForHost(ctx context.Context, symID string, h
 			}
 			for _, initiatorID := range initList.InitiatorIDs {
 				_, dirPort, _, err := splitFibreChannelInitiatorID(initiatorID)
-				if err != nil {
+				// Filter SCSI_FC ports; skip the port if it is not a SCSI_FC port
+				if err != nil || !slices.Contains(scsiPorts, dirPort) {
```

Comment:
Would have been better to have a utility function to check the port type.

Reply:
Initially, we planned to implement this approach. However, checking the port type for each port would require a separate API call, making it quite performance-intensive for Unisphere. So, we found an API that returns all the SCSI FC ports, and then we check if the initiator port is part of that list.

I have updated the code to increase readability:
```
		isSCSIFCPort := func(scsiPorts []string, port string) bool {
			return slices.Contains(scsiPorts, port)
		}
```
```
		// Filter SCSI_FC ports; skip the port if it is not a SCSI_FC port
		if err != nil || !isSCSIFCPort(scsiPorts, dirPort) {
			continue
		} 
```

Reply:
Port Group selection and creation is done only in FC scenario. For iSCSI and NVMe/TCP Port Group is passed during driver installation.